include/librealsense2/dds/topics/control/control-msg.h

```diff
@@ -0,0 +1,98 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+#include <string>
+#include <librealsense2/h/rs_sensor.h>
+#include "controlPubSubTypes.h"
+
+namespace librealsense {
+namespace dds {
+namespace topics {
+namespace device {
+class control
+{
+public:
+    using type = raw::device::controlPubSubType;
+
+    //Open uses a vector of profiles, to have constant message size we limit it
+    static const size_t MAX_OPEN_PROFILES = 8;
+
+#pragma pack( push, 1 )
+    enum class control_type : uint16_t
+    {
+        SENSOR_OPEN,
+        SENSOR_CLOSE,
+        MAX_CONTROL_TYPE
+    };
+
+    struct stream_profile
+    {
+        int16_t framerate;       // FPS
+        rs2_format format;       // Transfer as uint8_t?
+        rs2_stream type;         // Transfer as uint8_t?
+        int16_t width;           // Resolution width [pixels]
+        int16_t height;          // Resolution width [pixels]
+    };
+
+    struct sensor_open_msg
+    {
+        uint32_t message_id; // Running counter
+        int16_t sensor_uid;
+        stream_profile profiles[MAX_OPEN_PROFILES];
+    };
+
+    struct sensor_close_msg
+    {
+        uint32_t message_id; // Running counter
+        int16_t sensor_uid;
+    };
+
+
+#pragma pack( pop )
+
+    static std::string construct_topic_name( const std::string& topic_root )
+    {
+        return topic_root + "/control";
+    }
+
+    template<typename T>
+    static void construct_raw_message( control_type msg_id, const T& msg, raw::device::control& raw_msg )
+    {
+        raw_msg.id() = static_cast< int16_t >( msg_id );
+        raw_msg.size() = static_cast< uint32_t >( sizeof( msg ) );
+        raw_msg.raw_data().assign( reinterpret_cast< const uint8_t * >( &msg ),
+                                   reinterpret_cast< const uint8_t * >( &msg ) + raw_msg.size() );
+    }
+
+    control() = default;
+
+    control( const raw::device::control& main )
```

Comment:
If pass a `&&`, we can std::move...

Reply:
Hmm that's very interesting...
If true, this should be noted in the code because someone silly like me may try to optimize things and just cause problems.

But we use `take` and not `read` -- meaning it "‘removes’ it from the DataReader so it is no longer accessible" and so the prohibition that "any modifications made to the data will be seen the next time the same samples are read or taken" no longer applies.

I.e., the move is valid if you know what you're doing.

I wonder if the `take_next_sample` code, which currently sits with the user of the `device_info`, should be hidden inside `device_info` itself, so that it can control such things and not leave the (potentially faulty) logic to every client that comes around?

Reply:
Done
`device_info` handling `take` in [PR#10929](https://github.com/IntelRealSense/librealsense/pull/10929/files#diff-71f1d923f3f19865f4e77d070eebef4fd3851b74f06c0f8292d079306e8bb4d5)